src/DynamoPackagesUI/Utilities/PackageManagerCommands.cs

```diff
@@ -0,0 +1,485 @@
+            ResponseWithContentBody<dynamic> response = DPClient.ExecuteAndDeserializeDynamoRequest(req);
+            DownloadRequest = response.content;
+
+            PackageManagerRequest fileReq = new PackageManagerRequest(@"files/download?file_ids=" + temp[1] + "&asset_id=" + temp[0], Method.GET, true);
+            Response res = DPClient.ExecuteDynamoRequest(fileReq);
+            var pathToPackage = DPClient.GetFileFromResponse(res);
+            InstallPackage(pathToPackage);
+        }
+
+        public string GetCustomPathForInstall()
+        {
+            string downloadPath = string.Empty;
+
+            downloadPath = GetDownloadPath();
+
+            if (String.IsNullOrEmpty(downloadPath))
+                return string.Empty;
+
+            PackageInstallPath = downloadPath;
+
+            return downloadPath;
+        }
+
+        private string GetDownloadPath()
+        {
+            var args = new PackagePathEventArgs();
+
+            ShowFileDialog(args);
+
+            if (args.Cancel)
+                return string.Empty;
+
+            return args.Path;
+        }
+
+        private void ShowFileDialog(PackagePathEventArgs e)
+        {
+            string initialPath = dynamoViewModel.Model.PathManager.DefaultPackagesDirectory;
+
+            e.Cancel = true;
+
+            var dialog = new DynamoFolderBrowserDialog
+            {
+                // Navigate to initial folder.
+                SelectedPath = initialPath,
+                Title = "Install Path",
+                Owner = this.ParentWindow
+            };
+            Application.Current.Dispatcher.Invoke((Action)(() =>
+            {
+                if (dialog.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+                {
+                    e.Cancel = false;
+                    e.Path = dialog.SelectedPath;
+                }
+            }));
+        }
+
+        /// <summary>
+        /// Install button click event
+        /// </summary>
+        /// <param name="asset"></param>
+        /// <param name="version"></param>
+        /// <returns></returns>
+        public string PackageOnExecuted(dynamic asset, dynamic version)
+        {
+            string downloadPath = this.PackageInstallPath;
+
+            List<Tuple<dynamic, dynamic>> packageVersionData = new List<Tuple<dynamic, dynamic>>();
+
+            string msg = String.IsNullOrEmpty(downloadPath) ?
+                String.Format(Resources.MessageConfirmToInstallPackage, asset["asset_name"], version["version"]) :
+                String.Format(Resources.MessageConfirmToInstallPackageToFolder, asset["asset_name"], version["version"], downloadPath);
+
+            var result = MessageBox.Show(msg,
+                Resources.PackageDownloadConfirmMessageBoxTitle,
+                MessageBoxButton.OKCancel, MessageBoxImage.Question);
+
+            var pmExt = dynamoViewModel.Model.GetPackageManagerExtension();
```

Comment:
GetPackageManagerExtension is used at so many places. Can we store it as a field in the constructor of this class? Do you see the instance changing over the lifecycle of this class?